File: app/src/main/java/ru/ls/donkitchen/activity/receiptdetail/ReviewAdapter.java
package ru.ls.donkitchen.activity.receiptdetail;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RatingBar;
import android.widget.TextView;

import org.apache.commons.lang3.StringUtils;
import org.lucasr.twowayview.widget.TwoWayView;

import java.util.ArrayList;
import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;
import ru.ls.donkitchen.R;
import ru.ls.donkitchen.rest.model.response.ReviewListResult;

/**
 * @author Lord (Kuleshov M.V.)
 * @since 16.06.2015
 */
public class ReviewAdapter extends RecyclerView.Adapter<ReviewAdapter.ViewHolder> {
	public static class ViewHolder extends RecyclerView.ViewHolder {
		@Bind(R.id.date)
		TextView date;
		@Bind(R.id.rating)
		RatingBar rating;
		@Bind(R.id.author)
		TextView author;
		@Bind(R.id.comments)
		TextView comments;

		public ViewHolder(View view) {
			super(view);

			ButterKnife.bind(this, view);
		}
	}

	private final Context context;
	private final TwoWayView recyclerView;
	private final List<ReviewListResult.ReviewItem> items;

	public ReviewAdapter(Context context, TwoWayView recyclerView) {
		super();

		this.context = context;
		this.recyclerView = recyclerView;
		this.items = new ArrayList<>();
	}

	public void clear() {
		this.items.clear();
	}

	public void addAllItems(List<ReviewListResult.ReviewItem> items) {
		this.items.addAll(items);
	}

	public ReviewListResult.ReviewItem getItem(int position) {
		if (items != null && position < items.size()) {
			return items.get(position);
		}

		return null;
	}

	@Override
	public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		final View view = LayoutInflater.from(context).inflate(R.layout.list_item_review, parent, false);

		return new ViewHolder(view);
	}

	@Override
	public void onBindViewHolder(ViewHolder holder, int position) {
		ReviewListResult.ReviewItem item = items.get(position);

		holder.date.setText(item.date.toString("dd.MM.yyyy"));
		holder.rating.setRating(item.rating);
		holder.author.setText(item.userName);
		if (StringUtils.isEmpty(item.userName)) {
			holder.author.setVisibility(View.GONE);
		} else {
			holder.author.setVisibility(View.VISIBLE);
		}
		holder.comments.setText(item.comments);
		if (StringUtils.isEmpty(item.comments)) {
			holder.comments.setVisibility(View.GONE);
		} else {
			holder.comments.setVisibility(View.VISIBLE);
		}
	}

	@Override
	public int getItemCount() {
		return items.size();
	}
}

File: app/src/main/kotlin/ru/ls/donkitchen/activity/receiptdetail/ReviewAdapter.kt
package ru.ls.donkitchen.activity.receiptdetail

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.list_item_review.view.*
import ru.ls.donkitchen.R
import ru.ls.donkitchen.data.rest.response.ReviewListResult
import java.text.SimpleDateFormat
import java.util.*

/**
 *
 *
 * @author Lord (Kuleshov M.V.)
 * @since 17.10.16
 */
class ReviewAdapter(private val context: Context) : RecyclerView.Adapter<ReviewAdapter.ViewHolder>() {
    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        init {

        }
    }

    private val dateFormatter = SimpleDateFormat("dd.MM.yyyy", Locale.getDefault())

    private val items: MutableList<ReviewListResult.ReviewItem>?

    init {
        this.items = ArrayList<ReviewListResult.ReviewItem>()
    }

    fun clear() {
        this.items!!.clear()
    }

    fun addAllItems(items: List<ReviewListResult.ReviewItem>) {
        this.items!!.addAll(items)
    }

    fun getItem(position: Int): ReviewListResult.ReviewItem? {
        if (items != null && position < items.size) {
            return items[position]
        }

        return null
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.list_item_review, parent, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items!![position]

        holder.itemView.date.text = dateFormatter.format(item.date)
        holder.itemView.rating.rating = item.rating.toFloat()
        holder.itemView.author.text = item.userName
        if (TextUtils.isEmpty(item.userName)) {
            holder.itemView.author.visibility = View.GONE
        } else {
            holder.itemView.author.visibility = View.VISIBLE
        }
        holder.itemView.comments.text = item.comments
        if (TextUtils.isEmpty(item.comments)) {
            holder.itemView.comments.visibility = View.GONE
        } else {
            holder.itemView.comments.visibility = View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return items!!.size
    }
}
